Refactored for Chart to be single component. Style updates.

jordattebayo opened the pull request:
Incremental improvements - style updates, refactored createChart function to delete out of base.

jimmylee commented on pages/_/stats.js:
```diff
@@ -15,22 +15,22 @@ const randomDate = (start, end) => {
-  { id: 14, date: randomDate(new Date(2018, 0, 1), new Date()), category: "3", value: getRandomInt(1000) },
-];
+  let data = [
+    { id: 1, date: randomDate(new Date(2018, 0, 1), new Date()), category: "1", value: getRandomInt(1000) },
```
I think these dates should represent something linear right? I don't think this makes sense otherwise.

Jan 1, Jan, 2, Jan 3

for example

jordattebayo replied:
The dates are sorted before they are displayed so they are always linear. The randomDate function is a means to test the robustness of the component by throwing a variety of dates. I can revert it back to the linear hard coded sample data I had originally if you would prefer!

jimmylee commented on components/stats/Chart.js:
```diff
@@ -204,26 +267,124 @@ export default class Chart extends React.Component {
-      return <React.Fragment></React.Fragment>;
+      tickLines.push(tickLine);
     }
+    ReactDOM.render(tickLines, document.getElementById("tickContainer"));
```
You don't need `ReactDOM`, You can just use `this.state.ticks.map(each => {})` to achieve your goals and declare that inside of `render` I believe.

jimmylee commented on components/stats/Chart.js:
```diff
@@ -204,26 +267,124 @@ export default class Chart extends React.Component {
+        <circle key={index} cx={g.x} cy={g.y} r="2" css={STYLES_CHART_CIRCLE} />
+      );
+    });
+    ReactDOM.render(allCircles, document.getElementById("circles"));
```
`ReactDOM` isn't necessary, you should be able to get this all within `render` using `map` directly inside of a render function. For example, this is acceptable:

```
render() {
   return (
      <div>{allCircles}</div>
   )
}
```

jordattebayo replied:
It took some work but I refactored to remove reactDOM from the component and used your suggested method!

jimmylee reviewed (changes requested):
Making progress! I like it.

jordattebayo commented:
I think I've addressed your concerns, so please let me know if there is anything additional I need to update or change! Or if I should create a new PR.
